gbpservice/neutron/nsf/configurator/agents/vpn.py

```diff
@@ -0,0 +1,617 @@
+        else:
+            for svc_context in s2s_contexts:
+                svc_vendor = self._get_service_vendor(svc_context['service'])
+                self._resync_ipsec_conns(context, svc_vendor, svc_context)
+
+        try:
+            ssl_contexts = self.plugin_rpc.get_vpn_servicecontext(
+                context,
+                const.SERVICE_TYPE_OPENVPN,
+                filters={'status': ['PENDING_DELETE']})
+        except MessagingTimeout as err:
+            LOG.error("Failed in get_vpn_servicecontext for"
+                      " OPENVPN connections. Error: %s"
+                      % str(err).capitalize())
+        else:
+            for svc_context in ssl_contexts:
+                svc_vendor = self._get_service_vendor(svc_context['service'])
+                self._resync_openvpn_conns(context, svc_vendor, svc_context)
+
+    def update_conn_status(self,  context, svc_type, conn, status):
+        """
+        Driver will call this API to report
+        status of a connection - only if there is any change.
+        """
+        msg = ("Driver informing connection status "
+               "changed to %s" % status)
+        LOG.debug(msg)
+        if svc_type == const.SERVICE_TYPE_IPSEC:
+            vpnsvc_status = [{
+                'id': conn['vpnservice_id'],
+                'status':'ACTIVE',
+                'updated_pending_status':False,
+                'ipsec_site_connections':{
+                    conn['id']: {
+                        'status': status,
+                        'updated_pending_status': True}}}]
+
+        if svc_type == const.SERVICE_TYPE_OPENVPN:
+            vpnsvc_status = [{
+                'id': conn['vpnservice_id'],
+                'status':'ACTIVE',
+                'updated_pending_status':False,
+                'ssl_vpn_connections': {
+                    conn['id']: {
+                        'status': status,
+                        'updated_pending_status': True}}}]
+
+        self.plugin_rpc.update_status(context, vpnsvc_status)
+
+    def get_ipsec_contexts(self, context, tenant_id=None,
+                           vpnservice_id=None, conn_id=None,
+                           peer_address=None):
+        filters = {}
+        if tenant_id:
+            filters['tenant_id'] = tenant_id
+        if vpnservice_id:
+            filters['vpnservice_id'] = vpnservice_id
+        if conn_id:
+            filters['siteconn_id'] = conn_id
+        if peer_address:
+            filters['peer_address'] = peer_address
+
+        return self.plugin_rpc.\
+            get_vpn_servicecontext(
+                context,
+                const.SERVICE_TYPE_IPSEC, filters)
+
+    def get_ipsec_conns(self, context, filters):
+        return self.plugin_rpc.\
+            get_ipsec_conns(context, filters)
+
+
+def events_init(sc, drivers):
+    evs = [
+        Event(id='VPNSERVICE_UPDATED',
+              handler=VPNaasEventHandler(sc, drivers))]
```

Comment:
Change this as per latest configurator changes. WE need to pass notification queue object here.